### PR TITLE
Fixing issue #8367

### PR DESCRIPTION
**Pull request by sqdavid**
Summary of the changes
 - After Uri.TryCreate, concatenate the Fragment using the LocalPath separator.

Addresses #8367

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/RelatedAssemblyAttribute.cs [hunk 1]
```diff
@@ -117,6 +117,16 @@ internal static string GetAssemblyLocation(Assembly assembly)
         {
             if (Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var result) && result.IsFile)
             {
+                if (!string.IsNullOrWhiteSpace(result.Fragment))
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/RelatedAssemblyAttribute.cs]: I'd just do `Uri.TryCreate(assembly.CodeBase, UriKind.Absolute, out var result) && result.IsFile && string.IsNullOrEmpty(result.Fragment))`. Using `LocalPath` works well most of the times

Comment (sqdavid) [reply src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/RelatedAssemblyAttribute.cs]: Really? ok =] Fixing

**Comment by pranavkm** [comment]
@dotnet-bot build OSX 10.12 Release Build please

**Comment by sqdavid** [comment]
I’m using a MacOS XD.

**Comment by pranavkm** [comment]
Thanks for the PR

**Pull request merged**